### PR TITLE
add parakeet tutorial

### PR DESCRIPTION
**Pull request by Deep-unlearning**

### CODE REVIEW

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 1]
```diff
@@ -0,0 +1,361 @@
+# Fine-tuning Parakeet for ASR
+
+In this section, we will explore how to fine-tune NVIDIA's Parakeet CTC model for Automatic Speech Recognition (ASR) on a new language. Parakeet is a family of high-performance ASR models based on the FastConformer architecture, trained on over 64,000 hours of English speech data. We will fine-tune it on Spanish speech data to demonstrate how you can adapt this powerful English model to recognize speech in other languages.
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: "In this section, we will explore how to fine-tune NVIDIA's [Parakeet CTC](https://huggingface.co/nvidia/parakeet-ctc-1.1b) model for Automatic Speech Recognition (ASR) on a new language. Parakeet is a family of high-performance ASR models that use the [FastConformer](https://huggingface.co/papers/2305.05084) architecture for its encoder."

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+    "facebook/multilingual_librispeech", "spanish", split="test", streaming=True
+)
+```
+
+<Tip>
+    MLS supports 8 languages: German, Dutch, French, Spanish, Italian, Portuguese, Polish, and English.
+    Check the dataset card for more details: https://huggingface.co/datasets/facebook/multilingual_librispeech
+</Tip>
+
+We use streaming mode since MLS is a large dataset. Let's cast the audio to the correct sampling rate:
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: "We use [streaming mode](https://huggingface.co/blog/streaming-datasets) (`streaming=True`) since MLS is a large dataset.

We will also cast the audio to the desired sampling rate:"

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 3]
```diff
@@ -0,0 +1,361 @@
...
+
+```python
+mls_train = mls_train.cast_column("audio", Audio(sampling_rate=16000))
+mls_test = mls_test.cast_column("audio", Audio(sampling_rate=16000))
+```
+
+## Preprocessing
+
+Now let's load the processor:
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: "Now let's load the processor. One of its jobs is to prepare the audio before passing them to the model. For Parakeet, this _feature extraction_ consists of padding and computing the Mel Spectrogram."

File: chapters/en/_toctree.yml [hunk 4]
```diff
@@ -79,6 +79,8 @@
     title: Evaluation and metrics for speech recognition
   - local: chapter5/fine-tuning
     title: How to fine-tune an ASR system with the Trainer API
+  - local: chapter5/fine-tuning_parakeet
+    title: Fine-tuning Parakeet
```
Comment (ebezzam) [inline chapters/en/_toctree.yml:83]: How about these titles:

- Fine-tuning Whisper (OpenAI)
- Fine-tuning Parakeet CTC (NVIDIA) and using LoRA for efficient training

 Reasoning
- both use the Trainer API
- seeing company names can get people's attention
- pointing out LoRA usage in the Parakeet example

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 5]
```diff
@@ -0,0 +1,361 @@
...
+
+Make sure to log in to your Hugging Face account if you want to push your model to the Hub:
+
+```python
+from huggingface_hub import notebook_login
+
+notebook_login()
+```
+
+## The Dataset
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: Loading the Dataset?

Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: Loading the Dataset?

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 6]
```diff
@@ -0,0 +1,361 @@
...
+    references.append(sample["text"].lower())
+
+error_rate = wer(references, predictions)
+print(f"Word Error Rate: {error_rate:.2%}")
+```
+
+<Tip>
+For Spanish and other languages, it's good practice to normalize the text before computing WER. 
+This includes lowercasing, removing punctuation, and handling special characters consistently.
+</Tip>
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:263]: Worth mentioning/showing the one used in the Open ASR Leaderboard? https://github.com/huggingface/open_asr_leaderboard/tree/main/normalizer

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 7]
```diff
@@ -0,0 +1,361 @@
...
+
+```python
+import torch
+from jiwer import wer
+
+# Load your fine-tuned model
+from transformers import AutoModelForCTC, AutoProcessor
+
+model = AutoModelForCTC.from_pretrained("YOUR_USERNAME/parakeet-spanish-lora")
+processor = AutoProcessor.from_pretrained("YOUR_USERNAME/parakeet-spanish-lora")
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:230]: defining a variable `model_id` just above? so only needs to be set in one place (like you do in your example below)

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 8]
```diff
@@ -0,0 +1,361 @@
...
+```python
+# Test on a sample from MLS
+sample = next(iter(mls_test))
+print(f"Reference: {sample['text']}")
+
+transcription = transcribe(sample["audio"]["array"])
+print(f"Prediction: {transcription}")
+```
+
+## Using Other Languages
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:307]: Using Other Datasets?

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 9]
```diff
@@ -0,0 +1,361 @@
...
+    args=training_args,
+    train_dataset=mls_train,
+    eval_dataset=mls_test,
+    data_collator=data_collator,
+)
+
+trainer.train()
+```
+
+After training, save the model and push to the Hub:
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:208]: After training, we can save the model and push to the Hugging Face Hub:

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 10]
```diff
@@ -0,0 +1,361 @@
...
+```
+
+## Efficient Training with LoRA
+
+Fine-tuning a 1.1B parameter model can be memory-intensive. To fit this on a standard GPU, we can use Parameter-Efficient Fine-Tuning (PEFT) with LoRA:
+
+1.  **LoRA (Low-Rank Adaptation)**: Freezes most model weights and only trains small adapter layers.
+2.  **Gradient Checkpointing**: Trades compute for memory by re-computing activations during the backward pass.
+
+Let's configure the model for LoRA:
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:142]: Let's configure the model for LoRA with the `peft` library ([docs](https://huggingface.co/docs/peft/en/index)):

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 11]
```diff
@@ -0,0 +1,361 @@
+# Fine-tuning Parakeet for ASR
+
+In this section, we will explore how to fine-tune NVIDIA's Parakeet CTC model for Automatic Speech Recognition (ASR) on a new language. Parakeet is a family of high-performance ASR models based on the FastConformer architecture, trained on over 64,000 hours of English speech data. We will fine-tune it on Spanish speech data to demonstrate how you can adapt this powerful English model to recognize speech in other languages.
+
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:6]: What do think about linking a full Google colab notebook? so it's more convenient for people to run/download that rather than copying/pasting from here?

But still great to have this type of format for pedagogical purposes!

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 12]
```diff
@@ -0,0 +1,361 @@
...
+```python
+from huggingface_hub import notebook_login
+
+notebook_login()
+```
+
+## The Dataset
+
+For this example, we will use the Spanish subset of the [Multilingual LibriSpeech (MLS)](https://huggingface.co/datasets/facebook/multilingual_librispeech) dataset. MLS is a large-scale multilingual corpus derived from LibriVox audiobooks, containing thousands of hours of transcribed speech in 8 languages.
+
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:32]: We can conveniently load the MLS with the `datasets` library ([docs](https://huggingface.co/docs/datasets/en/index)):

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 13]
```diff
@@ -0,0 +1,361 @@
...
+)
+seed, buffer_size = 42, 10_000
+mls_train = mls_train.shuffle(seed=seed, buffer_size=buffer_size)
+mls_test = load_dataset(
+    "facebook/multilingual_librispeech", "spanish", split="test", streaming=True
+)
+```
+
+<Tip>
+    MLS supports 8 languages: German, Dutch, French, Spanish, Italian, Portuguese, Polish, and English.
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: MLS contains...

File: chapters/en/chapter5/fine-tuning_parakeet.mdx (hunk 1 above)
Comment (ebezzam) [reply chapters/en/chapter5/fine-tuning_parakeet.mdx]: Also there is a [multilingual Parakeet](https://huggingface.co/nvidia/parakeet-tdt-0.6b-v3), although TDT instead of CTC which is in the process of being added to Transformers.

So maybe we should reword the next part as people may wonder what's the point of finetuning for Spanish if there is already a Parakeet model that works for Spanish?

"The above linked Parakeet model has been trained on over 64,000 hours of English speech data, and we will fine-tune it on Spanish speech data to demonstrate how you can adapt this powerful English model to recognize speech in other languages.

While there exists a [multilingual Parakeet](https://huggingface.co/nvidia/parakeet-tdt-0.6b-v3) that supports 25 languages including Spanish, it uses a decoder not (yet) incorporated in Transformers. and cannot benefit from all the training goodies we will show below 🤗"

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 14]
```diff
@@ -0,0 +1,361 @@
...
+        # Process audio features and text labels
+        inputs = self.processor(
+            audio=audios,
+            text=texts,
+            sampling_rate=self.processor.feature_extractor.sampling_rate,
+        )
+        return inputs
+```
+
+## Training (Full Fine-tuning)
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: Training (Full Parameter Fine-tuning)

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 15]
```diff
@@ -0,0 +1,361 @@
...
+
+# Load your fine-tuned model
+model_id = "YOUR_USERNAME/parakeet-spanish-lora"
+model = AutoModelForCTC.from_pretrained(model_id, device_map="cuda")
+processor = AutoProcessor.from_pretrained(model_id)
+
+# Load an audio file
+import librosa
+
+audio_array, sr = librosa.load("path/to/spanish_audio.wav", sr=16000)
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:281]: Can you provide a path an actual audio? for user convenience

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 16]
```diff
@@ -0,0 +1,361 @@
...
+```
+
+MLS supports 8 languages: **German, Dutch, French, Spanish, Italian, Portuguese, Polish**, and **English**.
+
+For languages not in MLS or with non-Latin scripts (e.g., Arabic, Chinese, Japanese), you may need to consider:
+1. **Other datasets**: Use Common Voice or VoxPopuli for broader language coverage.
+2. **Vocabulary extension**: The Parakeet tokenizer is character-based but trained on English. You may need to extend it.
+3. **Phoneme-based approaches**: Convert text to phonemes using tools like `g2p` before training.
+4. **Alternative models**: Consider models pre-trained on multilingual data like Whisper for non-Latin scripts.
+
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx:341]: Can then finish off by mentioning fine-tuning on other datasets (which is why I suggested the broader title)

"Finally, check out the [Hugging Face Hub](https://huggingface.co/datasets?modality=modality:audio) for more datasets to fine-tune on!"

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 17]
```diff
@@ -0,0 +1,361 @@
...
+4. **Text normalization**: For non-English languages, ensure consistent text normalization (lowercase, punctuation removal, accent handling) during both training and evaluation.
+
+5. **Data augmentation**: Consider using speed perturbation, noise injection, or SpecAugment to improve robustness, especially for low-resource languages.
+
+6. **Combine datasets**: For better coverage, combine multiple datasets for your target language (e.g., MLS + Common Voice + VoxPopuli).
+
+## Summary
+
+In this tutorial, we demonstrated how to fine-tune NVIDIA's Parakeet model on Spanish speech data from Multilingual LibriSpeech. The key takeaways are:
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: "In this tutorial, we demonstrated how to fine-tune NVIDIA's Parakeet CTC model...

**Review by ebezzam — COMMENTED** [review]
@Deep-unlearning good stuff @Deep-unlearning 👏 

mainly nits for on wording and linking relevant material

File: chapters/en/chapter5/fine-tuning_parakeet.mdx [hunk 18]
```diff
@@ -0,0 +1,361 @@
+# Fine-tuning Parakeet for ASR
```
Comment (ebezzam) [inline chapters/en/chapter5/fine-tuning_parakeet.mdx]: Updating title accordingly

File: chapters/en/chapter5/fine-tuning_parakeet.mdx (hunk 1 above)
Comment (Deep-unlearning) [reply chapters/en/chapter5/fine-tuning_parakeet.mdx]: Thanks I mentionned multilingual parakeet!

File: chapters/en/chapter5/fine-tuning_parakeet.mdx (hunk 15 above)
Comment (Deep-unlearning) [reply chapters/en/chapter5/fine-tuning_parakeet.mdx:281]: I put a generic path since the audio file is expected to be local
